.github/workflows/tests.yml

```diff
@@ -0,0 +1,74 @@
+name: Run all tests
+
+# Run this workflow every time a new commit is pushed to the repository
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  run-all-tests:
+    name: Run all tests
+
+    # TODO: not sure what the best OS to run on is so ubuntu will do for now
+    runs-on: ubuntu-latest
+
+    strategy:
+      # Allows running the build multiple times with different configurations
+      matrix:
+        rust: ["stable", "beta"]
+        deny_warnings: ['', 1]
+
+        # TODO: I don't know how to make it run only the check_exercises.sh script
```

Comment:
Instead of grouping it in with this strategy, what if you created a new job, which just ran nightly?

Incidentally, the environment for nightly isn't `DENY_WARNINGS=1`; it's `BENCHMARK=1`.